File: multikube-rest-service/src/main/java/com/multikube_rest_service/dtos/requests/tenant/CreateNamespaceRequest.java
package com.multikube_rest_service.dtos.requests.tenant;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.Data;

/**
 * DTO for requesting the creation of a new tenant namespace.
 */
@Data
public class CreateNamespaceRequest {

    @NotBlank(message = "Namespace name cannot be blank.")
    @Pattern(regexp = "^[a-z0-9]([-a-z0-9]*[a-z0-9])?$", message = "Invalid Kubernetes namespace name. It must be DNS-compliant.")
    private String name;

    private String description;

    @NotNull(message = "A target cluster ID must be provided.")
    private Long clusterId;

    // Optional YAML manifest for ResourceQuota
    private String resourceQuotaYaml;

    // Optional YAML manifest for LimitRange
    private String limitRangeYaml;
}

File: multikube-ui/src/app/common/rest/types/tenant/requests/CreateNamespaceRequest.ts
// create-namespace-request.ts

export interface CreateNamespaceRequest {
    /**
     * The name for the new namespace.
     * - Cannot be blank.
     * - Must be a DNS-compliant name (e.g., 'my-awesome-namespace').
     * - Pattern: /^[a-z0-9]([-a-z0-9]*[a-z0-9])?$/
     */
    name: string;

    /**
     * An optional description for the namespace.
     */
    description?: string;

    /**
     * The ID of the target cluster where the namespace will be created.
     * - A target cluster ID must be provided.
     */
    clusterId: number;

    /**
     * Optional: A string containing the YAML manifest for a ResourceQuota object.
     */
    resourceQuotaYaml?: string;

    /**
     * Optional: A string containing the YAML manifest for a LimitRange object.
     */
    limitRangeYaml?: string;
}
